src/web-ui/buildspec.yml

```diff
@@ -2,25 +2,22 @@ version: 0.2
 phases:
   install:
     runtime-versions:
-      nodejs: 10
+      nodejs: 14
   pre_build:
     commands:
       - echo Installing NPM dependencies...
       - cd src/web-ui
-      - npm install
+      - npm ci
   build:
     commands:
-      - chmod a+x ./gen_env.sh
+      - chmod a+x *.sh
       - rm .env
       - ./gen_env.sh
       - cat .env
-      - npm run build
+      - ./build.sh
   post_build:
     commands:
-      - echo Uploading to ${WEB_BUCKET_NAME}
-      - aws s3 cp --recursive ./dist s3://${WEB_BUCKET_NAME}/ 
-      - aws s3 cp --cache-control="max-age=0, no-cache, no-store, must-revalidate" ./dist/index.html s3://${WEB_BUCKET_NAME}/
-      - aws cloudfront create-invalidation --distribution-id ${CLOUDFRONT_DIST_ID} --paths /index.html
+      - ./post_build.sh
```

Comment:
@james-jory XDN output folder is `src/web-ui/.xdn`. Do we need to add that to `artifacts` -> `base-directory`? If yes, how we can do that?

Reply:
Where does `src/web-ui/.xdn` need to be deployed? It looks like `post-build.sh` no longer copies the app to CloudFront so I assume it's not needed there. I need more info on how XDN needs to be deployed to answer your question.